trio/_core/_run.py

```diff
@@ -1576,8 +1636,16 @@ def reschedule(self, task, next_send=_NO_SEND):
             self.instruments.call("task_scheduled", task)
 
     def spawn_impl(
-        self, async_fn, args, nursery, name, *, system_task=False, context=None
-    ):
+        self,
+        # TODO: TypeVarTuple
+        async_fn: Callable[..., Awaitable[object]],
+        args: tuple[Any, ...],
+        nursery: Nursery | None,
+        name: object,
```

Comment:
`name: str | functools.partial | None`, and maybe also accepts `types.FunctionType`, but if unsure also add `object` as a catch all

Reply:
It accepts anything. For functions it uses the name, otherwise it `str()`ifys it.